.github/workflows/android.yml

```diff
@@ -26,16 +26,58 @@ jobs:
-      uses: gradle/gradle-build-action@v2
-      with:
-        arguments: build --info
-        build-root-directory: android
+      - name: Checkout
+        uses: actions/checkout@v2
+        with:
+          submodules: recursive
+          fetch-depth: 0 # Shallow clones should be disabled for a better relevancy of Sonar analysis
+
+      - name: Setup JDK
+        uses: actions/setup-java@v3
+        with:
+          distribution: "zulu"
+          java-version: "11"
+
+      - name: Cache Gradle
+        uses: actions/cache@v2
+        with:
+          path: |
+            ~/.gradle/caches
+            ~/.gradle/wrapper
+          key: gradle-${{ runner.os }}-${{ hashFiles('**/*.gradle*', '**/gradle-wrapper.properties') }}
+
+      - name: Grant execute permission for gradlew
+        run: |
+          chmod +x ./gradlew
+      - name: Build App
+        shell: bash
+        run: |
+          # use grep to remove log spam due to jetifier transformation, c.f. https://stackoverflow.com/a/52353446
+          # use set -o pipefail to keep error code from gradle, c.f. https://stackoverflow.com/a/37128767
+          ( set -o pipefail; ./gradlew assemble --parallel --build-cache --info | grep -vE "(^Transforming (.*) with (.*)Transform)|(^Caching disabled for (.*)Transform: (.*) because:)|(^  Caching not enabled.)|(^Skipping (.*)Transform: (.*) as it is up-to-date.)|(^Build cache key for (.*)Transform: (.*))" )
+      - name: Run tests
+        shell: bash
+        run: |
+          # use grep to remove log spam due to jetifier transformation, c.f. https://stackoverflow.com/a/52353446
+          # use set -o pipefail to keep error code from gradle, c.f. https://stackoverflow.com/a/37128767
+          ( set -o pipefail; ./gradlew build --parallel --build-cache --info | grep -vE "(^Transforming (.*) with (.*)Transform)|(^Caching disabled for (.*)Transform: (.*) because:)|(^  Caching not enabled.)|(^Skipping (.*)Transform: (.*) as it is up-to-date.)|(^Build cache key for (.*)Transform: (.*))" )
+      # - name: Generate Coverage Report
+        # shell: bash
+        # run: |
+          # # use grep to remove log spam due to jetifier transformation, c.f. https://stackoverflow.com/a/52353446
+          # # use set -o pipefail to keep error code from gradle, c.f. https://stackoverflow.com/a/37128767
+          # ( set -o pipefail; ./gradlew jacocoTestReport --parallel --build-cache --info | grep -vE "(^Transforming (.*) with (.*)Transform)|(^Caching disabled for (.*)Transform: (.*) because:)|(^  Caching not enabled.)|(^Skipping (.*)Transform: (.*) as it is up-to-date.)|(^Build cache key for (.*)Transform: (.*))" )
+      - name: Cache SonarCloud packages
+        uses: actions/cache@v2
+        with:
+          path: ~/.sonar/cache
+          key: ${{ runner.os }}-sonar
+          restore-keys: ${{ runner.os }}-sonar
+
+      # - name: SonarCloud Analysis
+        # env:
+          # SONAR_TOKEN: ${{ secrets.SONAR_TOKEN }}
```

Comment:
I setup a `SONAR_TOKEN_ANDROID` secret, so we should be able to run sonarcloud.

Reply:
Here is the gradle configuration: 

```
plugins {
  id "org.sonarqube" version "3.4.0.2513"
}

sonarqube {
  properties {
    property "sonar.projectKey", "hbt-android"
    property "sonar.organization", "dedis"
    property "sonar.host.url", "https://sonarcloud.io"
  }
}
```

Reply:
Done